src/main/java/httpserver/HttpHandler.java

```diff
@@ -0,0 +1,208 @@
+	 *                        keep the server from starting.
+	 */
+	public HttpHandler() { }
+
+
+	/**
+	 * Where the Handler handles the information given from the request and
+	 * based off of the paths specified in the Handler. <p>
+	 *
+	 * This can be overridden for more fine-grained handling. As is, it uses
+	 * the data behind the addGET, addPOST, and addDELETE methods for determining
+	 * the correct action to take. <p>
+	 *
+	 * If there is not exact match, the `*` path is used. If you don't have a `*`
+	 * catchall route, a 501 (Not implemented) is sent to the client.
+	 *
+	 * @param request     The incoming HttpRequest.
+	 * @param response    The outgoing HttpResponse, waiting to be filled by an
+	 *                    HttpHandler.
+	 *
+	 * @see HttpHandler#addGET
+	 * @see HttpHandler#addPOST
+	 * @see HttpHandler#addDELETE
+	 * @see HttpResponse#NOT_A_METHOD_ERROR
+	 */
+	public void handle(HttpRequest request, HttpResponse response) {
+		String httpRequestType = request.getRequestType().toUpperCase();
+		if (!routes.containsKey(httpRequestType)) {
+			response.message(404, "No " + httpRequestType + " routes exist.");
+			return;
+		}
+
+		MethodWrapper finalMethod = null;
+		int bestFit = 0;
+		for (MethodWrapper method : routes.get(httpRequestType)) {
+			if (method.matchesPerfectly(request.getSplitPath())) {
+				finalMethod = method;
+				break;
+			}
+
+			int testScore = method.howCorrect(request.getSplitPath());
+			if (testScore > bestFit) {
+				finalMethod = method;
+				bestFit = testScore;
+			}
+		}
+
+		if (finalMethod == null) {
+			response.message(501, HttpResponse.NOT_A_METHOD_ERROR);
+			return;
+		}
+
+		finalMethod.invoke(request, response, request.getSplitPath());
+	}
+
+	/**
+	 * Attach a method to a GET request at a path. <p>
+	 *
+	 * Methods are passed in as a String, and must be a member of the current
+	 * handler.<p>
+	 *
+	 * Path's should come in "/path/to/action" form. If the method requires
+	 * any parameters that aren't an HttpResponse, HttpRequest, or Map,
+	 * they should be included in the path, in the order they're
+	 * listed in the method header, in "{ClassName}" form. Example:
+	 * <code>/hello/{String}/{String}</code> is a good path. <p>
+	 *
+	 * Methods being passed in must accept an HttpResponse as their first
+	 * parameter. Methods may optionally accept an HttpRequest and a
+	 * Map&lt;String, String&gt; in that order (they may accept a Map but not an
+	 * HttpRequest, but if they accept both the HttpRequest must come first).
+	 *
+	 * Parameters following the above must be included in the java.lang library
+	 * and have a constructor that takes in a String.
+	 * Any other type of parameter will cause an exception to occur. <p>
+	 *
+	 * Additionally, primitives are not permited, because they're not classes in
+	 * the java.lang library. The three most common parameter types are String,
+	 * Integer, and Double.
```

Comment:
This is super out of date, and the meat of the documentation should be moved to `#addRoute`.